digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/pointasset/masstransitstop/ServicePointStopService.scala

```diff
@@ -0,0 +1,98 @@
+package fi.liikennevirasto.digiroad2.service.pointasset.masstransitstop
+
+import java.util.NoSuchElementException
+
+import fi.liikennevirasto.digiroad2._
+import fi.liikennevirasto.digiroad2.asset.{AdministrativeClass, BoundingRectangle, MassTransitStopAsset, Modification, PositionCoordinates, Property, PropertyValue, SimpleProperty}
+import fi.liikennevirasto.digiroad2.dao.Queries.updateAssetGeometry
+import fi.liikennevirasto.digiroad2.dao.{AssetPropertyConfiguration, Sequences, ServicePoint, ServicePointBusStopDao}
+import fi.liikennevirasto.digiroad2.oracle.OracleDatabase
+import fi.liikennevirasto.digiroad2.user.User
+
+case class ServicePointPublishInfo(asset: Option[ServicePoint], detachAsset: Seq[Long], attachedAsset: Seq[Long])
+case class PersistedPublicTransportStop(id: Long, nationalId: Long, stopTypes: Seq[Int], municipalityCode: Int, lon: Double, lat: Double, validityPeriod: Option[String],
+                                        vvhTimeStamp: Long, created: Modification, modified: Modification, propertyData: Seq[Property])
+
+class ServicePointStopService(eventbus: DigiroadEventBus) {
+  lazy val servicePointBusStopDao = new ServicePointBusStopDao()
+
+  def withDynSession[T](f: => T): T = OracleDatabase.withDynSession(f)
+  def withDynTransaction[T](f: => T): T = OracleDatabase.withDynTransaction(f)
+
+  def withId(id: Long)(query: String): String = {
+    query + s" where a.id = $id"
+  }
+
+  def create(lon: Double, lat: Double, properties: Seq[SimpleProperty], username: String, municipalityCode: Int): Long = {
+    val assetId = Sequences.nextPrimaryKeySeqValue
+    servicePointBusStopDao.insertAsset(assetId, lon, lat, username, municipalityCode)
+    servicePointBusStopDao.updateAssetProperties(assetId, properties)
+
```

Comment:
probably needs to be in a dynTransaction block

Reply:
Done